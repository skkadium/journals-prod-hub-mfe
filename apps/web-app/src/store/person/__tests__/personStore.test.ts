import { act, cleanup, renderHook } from '@testing-library/react'
import usePersonStore, { setPersons } from '../personStore'
import { mockedPersonDetails } from '../../../jest/jest.setup.unit'
describe('store/person/personStore', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let result: { current: any }
  beforeEach(() => {
    result = renderHook(() => usePersonStore()).result
  })
  afterEach(cleanup)

  it('should fetch initial person details state', () => {
    // Assert
    expect(result.current).toMatchSnapshot()
  })

  it('should update persons property', () => {
    expect(result.current.persons.length).toBe(0)

    // Act
    act(() => setPersons(mockedPersonDetails))

    // Assert
    expect(result.current.persons).toEqual(mockedPersonDetails)
  })
})
