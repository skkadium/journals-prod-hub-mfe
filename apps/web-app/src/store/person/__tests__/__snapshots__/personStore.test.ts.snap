// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/person/personStore should fetch initial person details state 1`] = `
{
  "persons": [],
}
`;
